index.js

```diff
@@ -201,11 +231,15 @@ module.exports = function AsyncAppendOnlyLog(filename, opts) {
       nextOffset = offset + recSize
     }
 
-    if (isBufferZero(dataBuf)) return [nextOffset, null]
-    else return [nextOffset, codec.decode(dataBuf)]
+    if (isBufferZero(dataBuf)) return [nextOffset, null, recSize]
+    else return [nextOffset, asRaw ? dataBuf : codec.decode(dataBuf), recSize]
   }
 
   function del(offset, cb) {
+    if (compaction) {
+      cb(new Error('Cannot delete while compaction is in progress'))
```

Comment:
I guess this should be added to errors.js?

Reply:
I see there are some others, what is the guiding principle if an error has an error code or not?